src/coreclr/tools/r2rdump/CoreDisTools.cs

```diff
@@ -1209,6 +1209,213 @@ private static bool IsAnotherRuntimeFunctionWithinMethod(int rva, RuntimeFunctio
             return false;
         }
 
+        /// <summary>
+        /// Improves disassembler output for LoongArch64.
+        /// </summary>
+        /// <param name="rtf">Runtime function</param>
+        /// <param name="imageOffset">Offset within the image byte array</param>
+        /// <param name="rtfOffset">Offset within the runtime function</param>
+        /// <param name="instruction">Textual representation of the instruction</param>
+        private void ProbeLoongArch64Quirks(RuntimeFunction rtf, int imageOffset, int rtfOffset, ref string instruction)
+        {
+            const int InstructionSize = 4;
+            uint instr = BitConverter.ToUInt32(_reader.Image, imageOffset + rtfOffset);
+
+            // The list of PC-relative instructions: BCond(BEQ, BNE, BLT[U], BGE[U]), BEQZ, BNEZ, BCEQZ, BCNEZ, B, BL, JIRL.
+
+            // Handle a B, BL, BCond(BEQ, BNE, BLT[U], BGE[U]), BZ(BEQZ, BNEZ, BCEQZ, BCNEZ) instruction
+            if (IsLoongArch64BCondInstruction(instr, out int offs) ||
+                IsLoongArch64BOrBlInstruction(instr, out offs) ||
+                IsLoongArch64BZInstruction(instr, out offs))
+            {
+                ReplaceRelativeOffset(ref instruction, rtf.StartAddress + rtfOffset + offs, rtf);
+            }
+            else if (IsLoongArch64JirlRAInstruction(instr, out uint rj, out int imm))
+            {
+                // Common Pattern:
+                //      pcaddu12i
+                //      ld.d
+                //      jirl  ra, rj, 0
+                //
+                //      pcaddu12i
+                //      addi.d
+                //      ld.d
+                //      jirl  ra, rj, 0
+                //  There may exist some irrelevant instructions between pcaddu12i and jirl.
+                //  We need to find relevant instructions based on rj to calculate the jump address.
+                uint register  = rj;
+                int  immediate = imm;
+                int  target    = 0;
```

Comment:
I think this variable is unused by the function

Reply:
Yes, thanks.